src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,329 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
                         : F.Block(F.Return(invocation));
         }
     }
+
+    internal class SynthesizedAsIntValueMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsIntValueMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi)
+            : base(module.SourceModule, pi,
+                   module.Compilation.GetSpecialType(SpecialType.System_Boolean),
+                   PrivateImplementationDetails.AsIntValueName)
+        {
+            this._module = module;
+            var compilation = _module.Compilation;
+            this.SetParameters(ImmutableArray.Create<ParameterSymbol>(
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Object), 0, RefKind.None, "o"),
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Int32), 1, RefKind.Out, "value")
+                ));
```

Comment:
This doesn't report use-site errors for types, are we certain we checked the types before? Please add a comment to that effect.